src/engine/storage/S3StorageEngine.js

```
@@ -0,0 +1,170 @@
+            }
+            throw error;
+        }
+    }
+
+    /**
+     * Get a model's index data
+     * @param {Model.constructor} constructor
+     * @throws MethodNotImplementedStorageEngineError
+     * @return Promise<object>
+     */
+    async getIndex(constructor) {
```

Biome (1.9.4)

[error] Do not shadow the global "constructor" property.

Consider renaming this variable. It's easy to confuse the origin of variables when they're named after a known global.

(lint/suspicious/noShadowRestrictedNames)